test/dynamo/test_exc.py

```diff
@@ -195,47 +195,132 @@ def fn001(x):
         inject_EVALUATE_EXPR_flip_equality_TESTING_ONLY=True,
         assume_static_by_default=False,
         translation_validation=True,
+        translation_validation_no_bisect=True,
         suppress_errors=False,
     )
     def test_trigger_on_error(self):
         from torch.fx.experimental.validator import ValidationException
 
         @torch.compile
-        def fn(x):
-            return x.reshape(-1, 4)
+        def fn(x, shape):
+            return x.split(shape)
 
         self.assertExpectedInlineMunged(
             ValidationException,
-            lambda: fn(torch.randn(20)),
+            lambda: fn(torch.randn(20), (5, 10, 5)),
             """\
 translation validation failed.
 
 Model:
+  ==> s0: 3
   ==> L['x'].storage_offset(): 0
-  ==> s0: 4
+  ==> L['shape'][0]: 0
+  ==> L['shape'][2]: 0
+  ==> s1: 0
   ==> L['x'].stride()[0]: 1
-  ==> L['x'].size()[0]: 4
+  ==> s2: 0
+  ==> L['shape'][1]: 0
+  ==> s3: 0
+  ==> L['x'].size()[0]: 3
 
 Assertions:
   ==> (== L['x'].size()[0] s0)
   ==> (> s0 1)
-  ==> (Not (And (< L['x'].size()[0] 4) (>= L['x'].size()[0] 0)))
+  ==> (== L['shape'][2] s3)
   ==> (True)
   ==> (== 0 L['x'].storage_offset())
+  ==> (== L['shape'][0] s1)
+  ==> (== L['shape'][1] s2)
   ==> (== 1 L['x'].stride()[0])
-  ==> (True)
 
 Target Expressions:
+  ==> (== L['x'].size()[0] s0)
+  ==> (== 0 s3)
+  ==> (>= 9223372036854775806 s1)
+  ==> (== 0 s2)
+  ==> (== 0 L['shape'][2])
+  ==> (>= 9223372036854775806 s3)
+  ==> (== 0 s1)
+  ==> (== 0 L['x'].storage_offset())
+  ==> (<= 0 s3)
+  ==> (<= 0 s2)
+  ==> (== 0 L['shape'][0])
+  ==> (> s0 0)
+  ==> (== 1 L['x'].stride()[0])
+  ==> (<= 2 s0)
+  ==> (== 0 L['shape'][1])
+  ==> (<= 0 s1)
+  ==> (>= 9223372036854775806 s2)
   ==> (>= 9223372036854775806 s0)
-  ==> (== 4 L['x'].size()[0])
+
+Failed Source Expressions:
+  ==> (!= 0 L['shape'][1])
+  ==> (!= 0 L['shape'][0])
+  ==> (== (+ L['shape'][0] L['shape'][1] L['shape'][2]) L['x'].size()[0])
+  ==> (!= 0 L['shape'][2])""",
+        )
+
+    @skipIf(not TEST_Z3, "z3 not installed")
+    @torch._dynamo.config.patch(
+        inject_EVALUATE_EXPR_flip_equality_TESTING_ONLY=True,
+        assume_static_by_default=False,
+        translation_validation=True,
+        suppress_errors=False,
+    )
+    def test_trigger_bisect_on_error(self):
+        from torch.fx.experimental.validator import ValidationException
+
+        @torch.compile
+        def fn(x, shape):
+            return x.split(shape)
+
+        self.assertExpectedInlineMunged(
+            ValidationException,
+            lambda: fn(torch.randn(20), (5, 10, 5)),
+            """\
+translation validation failed.
```

Comment:
How exactly am I supposed to interpret the message here?

Reply:
The problem that TV tries to solve is: _"Given **Assertions**, is there any **Model** that satisfies **Target Expressions**, but doesn't satisfy the source expressions?"_. The **Faiiled Source Expressions** represents an affirmative response, showing which source expressions that failed.

Reply:
Oh, I see. I thought the smaller problem instance would be a nice win already, e.g. easier to read and reason about. But that makes sense.

Reply:
Another thing I thought was that it would be nice to know under what program FX node we failed. But, is there any way to get this information at this point (inside `ShapeEnv`)?

Reply:
Yeah, the stack trace on the bad guard would be pretty useful.

Reply:
What do you think of this?

Reply:
Yes, this is better.